File: backend/schemas/project.py
from typing import Optional
from backend.schemas.base import BaseSchema


class ProjectCreateRequest(BaseSchema):
    name: str
    description: Optional[str] = None


class ProjectUpdateRequest(BaseSchema):
    name: str
    description: Optional[str] = None


class ProjectResponse(BaseSchema):
    id: int
    name: str
    description: Optional[str] = None
    status: str = "DRAFT"

File: frontend/src/types/project.ts
export interface Project {
  id: number;
  name: string;
  description?: string | null;
  status: string; // "DRAFT" | "SELECTED"
}

export interface ProjectCreateRequest {
  name: string;
  description?: string | null;
}

export interface ProjectUpdateRequest {
  name: string;
  description?: string | null;
}
